### PR TITLE
980 - fix möglicher Nullpointer beim Logger des Usernamens

### LINKED ISSUE #980: Als Benutzer möchte ich mich ausloggen können
# Todos
- [x] nach Release finales openAPI-Doc in wls-gui hinterlegen: PR #2209
- [x] Docker-compose auf wls-gateway umstellen: PR #2211
- [x] https://it-at-m.github.io/Wahllokalsystem/technik/guides/new-microservice/new-service-backend.html
- [x] Infrastruktur umstellen - PR gestellt
- [x] experimental-Branch löschen
- [x] bhb: bei AuthService vermerken dass logout-URL gesetzt sein muss und zum Login passen muss
- [x] Label für API-Gateway erstellen und im Workflow konfigurieren #2208 
- [x] Support für alte adminGui umsetzen

# Akzeptanzkriterien
- [ ] ein Logout ist über das WLS-UI möglich
- [ ] nach einem Logout kann man sich mit einem anderen User einloggen (ohne das Browser-Fenster schließen zu müssen)
- [ ] die korrekte Komplexität wurde dokumentiert

### PR DESCRIPTION
**Pull request by MrSebastian**
# Beschreibung:

Korrektur des Fehlers aus dem [Kommentar](https://github.com/it-at-m/Wahllokalsystem/issues/980#issuecomment-3570965235) des Issues

### Backend ###
- [x] [Naming Conventions][naming-conventions-link] beachtet
- [x] ~~Doku aktualisiert~~
- [X] ~~Swagger-API vollständig~~
- [X] Unit-Tests gepflegt
- [X] ~~Integrationstests gepflegt~~
- [X] ~~Beispiel-Requests gepflegt~~
- [X] ~~Texte auf Rechtschreibung und Grammatik geprüft~~

# Referenzen[^1]:

Verwandt mit Issue #

Closes #980

> [^1]: _Nicht zutreffende Referenzen vor dem Speichern entfernen_

[naming-conventions-link]: https://it-at-m.github.io/Wahllokalsystem/technik/naming_conventions/
[fachliche-beschreibung-link]: https://it-at-m.github.io/Wahllokalsystem/about/#fachliche-anforderungen
[ui-ux-adrs-link]: https://it-at-m.github.io/Wahllokalsystem/technik/adr/ui/

## Release Notes

* **Refactoring**
  * Centralized username extraction logic across gateway and authentication services, improving consistency when handling different authentication types (JWT tokens and traditional credentials) in logging.

* **Tests**
  * Added comprehensive unit test coverage for authentication utilities, ensuring proper username extraction for various authentication scenarios and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            void should_returnNull_when_jwtAuthenticationTokenIsGivenButHasNoUsername() {
+                val authentication = new JwtAuthenticationToken(
+                        new Jwt("tokenValue", Instant.now(), Instant.MAX, Map.of("key", "value"), Map.of("notTheUsernameClaim", USERNAME)));
+
+                val result = AuthUtils.getUsername(authentication);
+                Assertions.assertThat(result).isNull();
+            }
+
+            @Test
+            void should_returnUsername_when_usernamePasswordAuthenticationToken() {
```
Comment (vjohnslhm) [inline wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java]:
```suggestion
            void should_returnUsername_when_usernamePasswordAuthenticationTokenIsGiven() {
```

File: wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                val authentication = new UsernamePasswordAuthenticationToken("username", "password");
```
Comment (vjohnslhm) [inline wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java]: wenn sich hier `"username"` ändert läuft der test nicht mehr deswegen würde ich auch hier mit der konstante arbeiten

File: wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    @Override
+                    public void setAuthenticated(boolean isAuthenticated) {
+
+                    }
+
+                    @Override
+                    public String getName() {
+                        return "";
+                    }
+                });
```
Comment (vjohnslhm) [inline wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java]: das würde ich für bessere lesbarkeit nach unten ins file auslagern

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/security/AuthUtilsTest.java
Comment (vjohnslhm) [inline wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/security/AuthUtilsTest.java:1]: gleiche anmerkungen wie beim anderen testfile. bitte 
- testname anpassen
- konstante verwenden
- create Authentication auslagern

**Review by vjohnslhm — CHANGES_REQUESTED** [review]

File: wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java (hunk 1 above)
Comment (MrSebastian) [reply wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java]: umgesetzt mit b80784e29279b8c6eae96eb50ebf08837dfedd1f

File: wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java (hunk 2 above)
Comment (MrSebastian) [reply wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java]: umgesetzt mit b80784e29279b8c6eae96eb50ebf08837dfedd1f

File: wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java (hunk 3 above)
Comment (MrSebastian) [reply wls-api-gateway/src/test/java/de/muenchen/refarch/gateway/security/AuthUtilsTest.java]: umgesetzt mit b80784e29279b8c6eae96eb50ebf08837dfedd1f

File: wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/security/AuthUtilsTest.java
Comment (MrSebastian) [reply wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/security/AuthUtilsTest.java:1]: umgesetzt mit b80784e29279b8c6eae96eb50ebf08837dfedd1f

**Review by vjohnslhm — DISMISSED** [review]

**Review by vjohnslhm — APPROVED** [review]
